File: member-service/src/main/java/com/hedvig/memberservice/aggregates/LivingAddress.java
package com.hedvig.memberservice.aggregates;

import java.util.Objects;
import lombok.Value;

@Value
public class LivingAddress {

  /** Represents Street, number, entrance */
  String street;

  String city;
  String zip;
  String apartmentNo;
  Integer floor;

  public boolean needsUpdate(
      String street, String city, String zipCode, String apartmentNo, Integer floor) {
    if (street == null && city == null && zipCode == null && apartmentNo == null && floor == null) {
      return false;
    }
    return !Objects.equals(this.street, street)
        || !Objects.equals(this.city, city)
        || !Objects.equals(this.zip, zipCode)
        || !Objects.equals(this.apartmentNo, apartmentNo)
        || !Objects.equals(this.floor, floor);
  }
}

File: member-service/src/main/java/com/hedvig/memberservice/aggregates/LivingAddress.kt
package com.hedvig.memberservice.aggregates

class LivingAddress(
    /** Represents Street, number, entrance  */
    val street: String?,
    val city: String?,
    val zip: String?,
    val apartmentNo: String?,
    val floor: Int?
) {
    fun needsUpdate(
        street: String?, city: String?, zipCode: String?, apartmentNo: String?, floor: Int?): Boolean {
        return if (street == null && city == null && zipCode == null && apartmentNo == null && floor == null) {
            false
        } else this.street != street
            || this.city != city
            || zip != zipCode
            || this.apartmentNo != apartmentNo
            || this.floor != floor
    }

    override fun equals(other: Any?): Boolean {
        if (other !is LivingAddress)
            return false
        if (street != other.street)
            return false
        if (city != other.city)
            return false
        if (zip != other.zip)
            return false
        if (apartmentNo != other.apartmentNo)
            return false
        if (floor != other.floor)
            return false
        return true
    }
}
